File: seon/level2/12981.js
function solution(n, words) {
  const check = {};
  let currentUser = 1;
  let cnt = 1;
  let idx = 0;

  for (let word of words) {
    if (check[word]) return [currentUser, cnt];
    else if (
      words[idx - 1] &&
      words[idx - 1][words[idx - 1].length - 1] !== word[0]
    )
      return [currentUser, cnt];
    else {
      check[word] = true;

      if (currentUser === n) {
        currentUser = 1;
        cnt += 1;
      } else currentUser += 1;

      idx += 1;
    }
  }

  return [0, 0];
}

File: eunna/level2/12981.py
def solution(n, words):
    answer = [0, 0]

    for i in range(len(words) - 1):
        for j in range(i - 1, -1, -1):
            if words[i] == words[j]:
                answer = [i % n + 1, i // n + 1]
                return answer
        if words[i][-1] != words[i + 1][0]:
            answer = [(i + 1) % n + 1, (i + 1) // n + 1]
            return answer

    for j in range(len(words) - 2, -1, -1):
        if words[len(words) - 1] == words[j]:
            answer = [(len(words) - 1) % n + 1, (len(words) - 1) // n + 1]
            return answer

    return answer
